File: 23-Observer/src/go/window_display.go
package main

import "fmt"

// WindowDisplay is a Concrete Observer — a dashboard that reacts to temperature.
type WindowDisplay struct{}

func (w *WindowDisplay) Update(temperature float64) {
	advice := "keep them closed"
	if temperature > 25 {
		advice = "open the windows"
	}
	fmt.Printf("  [Window] %.1f°C — %s\n", temperature, advice)
}

File: 23-Observer/src/rust/window_display.rs
use crate::observer::Observer;

/// Concrete Observer — a dashboard that reacts to temperature.
pub struct WindowDisplay;

impl Observer for WindowDisplay {
    fn update(&self, temperature: f64) {
        let advice = if temperature > 25.0 { "open the windows" } else { "keep them closed" };
        println!("  [Window] {:.1}°C — {}", temperature, advice);
    }
}
